types/cookieclicker/index.d.ts

```diff
@@ -0,0 +1,2973 @@
+
+        tieredAchievs: Array<TieredAchievementClass<number>>;
+
+        tieredUpgrades: Array<TieredUpgradeClass<number>>;
+        /**
+         * Generates a tooltip to show on the shop listing.
+         * @return A string with the html elements
+         */
+        tooltip: () => string;
+        /**
+         * Total cookies produced by the building
+         */
+        totalCookies: number;
+
+        vanilla: PseudoBoolean;
+    }
+    export { GameObject as Object };
+
+    export interface MinigameObject<T extends Minigame> extends GameObject {
+        minigame: T;
+    }
+
+    export function DrawBuildings(): void;
+    export function sortSprites(a: any, b: any): number;
+    export function sortSpritesById(a: any, b: any): number;
+
+    export function modifyBuildingPrice(building: string, price: number): number;
+
+    export function storeBulkButton(id: number): void;
+
+    export function BuildStore(): void;
+
+    export function RefreshStore(): void;
+
+    export function ComputeCps(base: number, mult: number, bonus: number): number;
+    export function isMinigameReady(me: GameObject): void;
+    export let scriptBindings: undefined[] & Record<string, GameObject>;
+
+    export function LoadMinigames(): void;
+
+    export function scriptLoaded(who: GameObject, script: string): void;
+
+    export function magicCpS(what: unknown): number;
+    export let SpecialGrandmaUnlock: number;
+    export let foolObjects: FoolBuilding[];
+
+    export function ClickProduct(what: GameObject): void;
+
+    export function mutedBuildingTooltip(id: number): () => string;
+    export let upgradesToRebuild: number;
+    export let Upgrades: undefined[] & Record<string, Upgrade>;
+    export let UpgradesById: Upgrade[];
+    export let UpgradesN: number;
+    export let UpgradesInStore: Upgrade[];
+    export let UpgradesOwned: number;
+
+    export type UpgradePool = '' | 'prestige' | 'tech' | 'cookie' | 'debug' | 'toggle' | 'prestigeDecor' | 'unused';
+    export class Upgrade {
+        /**
+         * Creates a new generic upgrade
+         * @param name The name of the upgrade
+         * @param desc The description of the upgrade
+         * @param price The price in cookies for the upgrade
+         * @param icon The Icon to use for the upgrade
+         * @param buyFunction The function which gets called when the upgrade bought
+         */
+        constructor(name: string, desc: string, price: number, icon: Icon, buyFunction?: () => void);
+
+        /**
+         * The description of the upgrade without auto-adjusted text
+         */
+        baseDesc: string;
+        /**
+         * The description of the upgrade with auto-adjusted text
+         */
+        desc: string;
+        /**
+         * The function to generate the upgrade descroption
+         */
+        descFunc?: () => string;
+        /**
+         * 	The price of the upgrade without the cost multipliers
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/cookieclicker/index.d.ts;
  Line: 2176;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.